### PR TITLE
better sudo/su deprecation on config entries

### LINKED ISSUE #32130: Deprecation warnings are insufficiently detailed to fix the configuration
##### ISSUE TYPE

 - Documentation Report

##### COMPONENT NAME

Deprecation warnings

##### ANSIBLE VERSION

```
ansible 2.4.0.0
  config file = /work/ansible/configuration/ansible.cfg
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]

```

##### CONFIGURATION

```
[DEPRECATION WARNING]: DEFAULT_SUDO_FLAGS option, In favor of become which is a generic framework . This feature will be removed in version 2.8. Deprecation warnings can be 
disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: DEFAULT_SUDO_EXE option, In favor of become which is a generic framework . This feature will be removed in version 2.8. Deprecation warnings can be 
disabled by setting deprecation_warnings=False in ansible.cfg.
ANSIBLE_NOCOLOR(/work/ansible/configuration/ansible.cfg) = True
ANSIBLE_NOCOWS(/work/ansible/configuration/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/work/ansible/configuration/ansible.cfg) = -o ControlMaster=auto -o ControlPersist=60s -o ForwardAgent=yes
CACHE_PLUGIN(/work/ansible/configuration/ansible.cfg) = jsonfile
CACHE_PLUGIN_CONNECTION(/work/ansible/configuration/ansible.cfg) = /home/user/.ansible/cachedir
CACHE_PLUGIN_TIMEOUT(/work/ansible/configuration/ansible.cfg) = 86400
DEFAULT_ACTION_PLUGIN_PATH(/work/ansible/configuration/ansible.cfg) = [u'/usr/share/ansible_plugins/action_plugins']
DEFAULT_ASK_PASS(/work/ansible/configuration/ansible.cfg) = False
DEFAULT_CALLBACK_PLUGIN_PATH(/work/ansible/configuration/ansible.cfg) = [u'/usr/share/ansible_plugins/callback_plugins']
DEFAULT_CONNECTION_PLUGIN_PATH(/work/ansible/configuration/ansible.cfg) = [u'/usr/share/ansible_plugins/connection_plugins']
DEFAULT_FILTER_PLUGIN_PATH(/work/ansible/configuration/ansible.cfg) = [u'/work/ansible/filter_plugins']
DEFAULT_FORKS(/work/ansible/configuration/ansible.cfg) = 5
DEFAULT_GATHERING(/work/ansible/configuration/ansible.cfg) = implicit
DEFAULT_HOST_LIST(/work/ansible/configuration/ansible.cfg) = [u'/work/ansible/configuration/hosts']
DEFAULT_LOOKUP_PLUGIN_PATH(/work/ansible/configuration/ansible.cfg) = [u'/usr/share/ansible_plugins/lookup_plugins']
DEFAULT_MANAGED_STR(/work/ansible/configuration/ansible.cfg) = Ansible managed: {file} modified on %Y-%m-%d %H:%M:%S by {uid} on {host}
DEFAULT_MODULE_LANG(/work/ansible/configuration/ansible.cfg) = C
DEFAULT_POLL_INTERVAL(/work/ansible/configuration/ansible.cfg) = 15
DEFAULT_REMOTE_TMP(/work/ansible/configuration/ansible.cfg) = $HOME/.ansible/tmp
DEFAULT_SCP_IF_SSH(/work/ansible/configuration/ansible.cfg) = True
DEFAULT_SUDO_EXE(/work/ansible/configuration/ansible.cfg) = sudo
DEFAULT_SUDO_FLAGS(/work/ansible/configuration/ansible.cfg) = -n
DEFAULT_TIMEOUT(/work/ansible/configuration/ansible.cfg) = 10
DEFAULT_TRANSPORT(/work/ansible/configuration/ansible.cfg) = ssh
DEFAULT_VARS_PLUGIN_PATH(/work/ansible/configuration/ansible.cfg) = [u'/usr/share/ansible_plugins/vars_plugins']
RETRY_FILES_ENABLED(/work/ansible/configuration/ansible.cfg) = False

```
##### OS / ENVIRONMENT

RHEL 7.4

##### SUMMARY

The deprecation warnings, as seen in the config dump above, are confusing and could be improved,
Several points:

- They reference a shell variable, with no indication where the error lies, only how to hide the warning. Ie, it's not even possible to grep around to find exactly what's amiss.
- The current documentation, at the time I'm writing this, does not indicate that this configuration is deprecated (quoted below in full, just in case it changes soon). http://docs.ansible.com/ansible/latest/intro_configuration.html#sudo-flags
- The devel documentation does not help, it states that the variable is deprecated, but gives no clear way to replace it (it only says to use 'become', but you can't set a flags to sudo in 'become' itself) https://docs.ansible.com/ansible/devel/config.html

I was pointed to 'become_flags', but this is not an exact equivalent to 'sudo_flags'.
Particularly, I understand that if 'become_flags' is set globally for sudo-specific parameters, it will have to be reset in case a different method is used.
And that raise a question, since the defaults, if any, for each methods are not obvious anymore, it makes that reset error-prone in case the right defaults are not used.

I suggest that documentation be improved to explain in more details how 'sudo_flags' is to be replaced before it's fully removed.

Quote of the 'latest' documentation at the time of writing:
```
sudo_flags

Additional flags to pass to sudo when engaging sudo support. The default is ‘-H -S -n’ which sets the HOME environment variable, prompts for passwords via STDIN, and avoids prompting the user for input of any kind. Note that ‘-n’ will conflict with using password-less sudo auth, such as pam_ssh_agent_auth. In some situations you may wish to add or remove flags, but in general most users will not need to change this setting::

sudo_flags=-H -S -n
```

Quote of the 'devel' documentation at the time of writing:
```
DEFAULT_SUDO_FLAGS
Description:    Flags to pass to “sudo”
Default:        -H -S -n
Ini Section:    defaults
Ini Key:        sudo_flags
Environment:    ANSIBLE_SUDO_FLAGS
Deprecated in:  2.8
Deprecated detail:
        In favor of become which is a generic framework
Deprecated alternatives:
        become
```

##### STEPS TO REPRODUCE

Set 'sudo_flags' in ansible.cfg.

##### EXPECTED RESULTS

A sufficiently detailed set of steps to replace the deprecated options with the current supported ones without disruption to the existing playbooks.

##### ACTUAL RESULTS

A too terse message that's not very helpful in finding and fixing the potential issue.

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

fixes #32130

##### ISSUE TYPE

 - Docs Pull Request

##### COMPONENT NAME

become
##### ANSIBLE VERSION

```
2.5
```

### CODE REVIEW

File: lib/ansible/config/base.yml [hunk 1]
```diff
@@ -373,7 +373,7 @@ DEFAULT_ASK_SU_PASS:
   name: Ask for the su password
   default: False
   deprecated:
-    why: In favor of become which is a generic framework
+    why: In favor of Ansible Become which is a generic framework, see becoem_ask_pass
```
Comment (gundalow) [inline lib/ansible/config/base.yml]: typo `become_ask_pass`

File: lib/ansible/config/base.yml [hunk 2]
```diff
@@ -1044,7 +1044,7 @@ DEFAULT_SU_FLAGS:
   name: su flags
   default: ''
   deprecated:
-    why: In favor of become which is a generic framework
+    why: In favor of Ansible Become which is a generic framework, see becoem_flags
```
Comment (gundalow) [inline lib/ansible/config/base.yml]: typo `become_flags`

**Review by gundalow — COMMENTED** [review]
Nice

File: lib/ansible/config/base.yml (hunk 2 above)
Comment (bcoca) [reply lib/ansible/config/base.yml]: trying to set a new trend here ...

**Review by dharmabumstead — APPROVED** [review]
OK with edits. Thanks @bcoca!

**Pull request merged**